applications/CompressiblePotentialFlowApplication/python_scripts/stochastic_potential_flow_response.py

```diff
@@ -0,0 +1,332 @@
+import KratosMultiphysics
+from KratosMultiphysics import Parameters, Logger
+import KratosMultiphysics.CompressiblePotentialFlowApplication as KCPFApp
+from KratosMultiphysics.response_functions.response_function_interface import ResponseFunctionInterface
+import KratosMultiphysics.CompressiblePotentialFlowApplication.potential_flow_analysis as potential_flow_analysis
+
+# Import Kratos, XMC, PyCOMPSs API
+import KratosMultiphysics.MultilevelMonteCarloApplication
+import xmc
+import xmc.methodDefs_momentEstimator.computeCentralMoments as mdccm
+from exaqute import get_value_from_remote
+import json
+
+def _GetModelPart(model, solver_settings):
+    #TODO can be removed once model is fully available
+    model_part_name = solver_settings["model_part_name"].GetString()
+    if not model.HasModelPart(model_part_name):
+        model_part = model.CreateModelPart(model_part_name, 2)
+        domain_size = solver_settings["domain_size"].GetInt()
+        if domain_size < 0:
+            raise Exception('Please specify a "domain_size" >= 0!')
+        model_part.ProcessInfo.SetValue(KratosMultiphysics.DOMAIN_SIZE, domain_size)
+    else:
+        model_part = model.GetModelPart(model_part_name)
+
+    return model_part
+
+class AdjointResponseFunction(ResponseFunctionInterface):
```

Comment:
@armingeiser, to clarify, in this "AdjointResponseFunction" XMC is run at each `InitializeSolutionStep()` and the value and gradient for the step is retrieved from the XMC data structures. 

Inside XMC, a custom analysis stage is used to solve the primal+adjoint, which is defined below: `SimulationScenario`
https://github.com/KratosMultiphysics/Kratos/blob/3269f555bdcb246b3ebfb95ca6c634f0e6b80178/applications/CompressiblePotentialFlowApplication/python_scripts/stochastic_potential_flow_response.py#L201

Using this custom analysis is specified in the parameters:
https://github.com/KratosMultiphysics/Kratos/blob/3269f555bdcb246b3ebfb95ca6c634f0e6b80178/applications/CompressiblePotentialFlowApplication/tests/stochastic_body_fitted_opt/xmc_parameters.json#L54